.circleci: Switch to dynamic MAX_JOBS

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44729 .circleci: Switch to dynamic MAX_JOBS**

Switches our MAX_JOBS from a hardcoded value to a more dynamic value so
that we can always utilize all of the core that are available to us

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23759643](https://our.internmc.facebook.com/intern/diff/D23759643)

seemethere committed:
.circleci: Switch to dynamic MAX_JOBS

Switches our MAX_JOBS from a hardcoded value to a more dynamic value so
that we can always utilize all of the core that are available to us

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

walterddr commented on .circleci/scripts/binary_linux_build.sh:
```diff
@@ -7,7 +7,7 @@ source /env
 # This script is run inside Docker.2XLarge+ container that has 20 CPU cores
 # But ncpu will return total number of cores on the system
-export MAX_JOBS=18
+export MAX_JOBS=$(( $(nproc) - 2 ))
```
I thought `nproc` was suppose to return # of available CPU on the host instead of the container? @malfet

seemethere replied:
We had a conversation about it and we think the issue with the jobs stemmed from the use of `stdbuf` and not necessarily from the `nproc` invocation

malfet replied:
@seemethere can you please delete comments on lines 8-9 then?
@walterddr yes, that is correct, but actually spawning more build jobs than number of cores is fine.

malfet reviewed (approved):
LGTM provided that build time is lower across linux platforms than the current build time.

seemethere committed:
Update on ".circleci: Switch to dynamic MAX_JOBS"


Switches our MAX_JOBS from a hardcoded value to a more dynamic value so
that we can always utilize all of the core that are available to us

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

malfet commented:
@seemethere alternatively, perhaps we could try installing `ninja` on the nightly docker images and benefit from its autoscale feature.

xuzhao9 reviewed (approved):
LGTM. According to [this issue](https://github.com/moby/moby/issues/20770#issuecomment-190540965), the available cpus within a docker from the file `/sys/fs/cgroup/cpuset/cpuset.cpus`.
